graph/src/util/log.rs

```diff
@@ -67,5 +73,20 @@ pub fn register_panic_hook(panic_logger: Logger) {
                 );
             }
         };
+
+        // Send a shutdown signal to main which will attempt to cleanly shutdown the runtime
+        // After sending shutdown, the thread sleeps for 3 seconds then forces the process to
+        // exit because the shutdown is not always able to cleanly exit all workers
+        shutdown_sender
+            .lock()
+            .unwrap()
+            .take()
+            .expect("Shutdown signal already sent")
```

Comment:
So, the mutex can be unwrapped, but I don't think it's the best idea to do `expect` on the `take()`, I'm not sure but it may be the case that multiple panics may call this hook simultaneously or in sequence.